udata/core/dataset/models.py

```diff
@@ -897,6 +900,9 @@ def get(cls, id_or_slug):
     def add_resource(self, resource):
         """Perform an atomic prepend for a new resource"""
         resource.validate()
+        if resource.id in [r.id for r in self.resources]:
+            raise RuntimeError("Cannot add resource with already existing ID")
```

Comment:
Should it be a RuntimeError by the way?

Reply:
I'm not sure about the policy on errors. Do we create a new error type? Do we use something generic?

Reply:
I would say we could use a `MongoEngineValidationError` as you did above?

Reply:
Changed in 35e775e1186ade0b607a3610fa251d262a8f487e